### DISCUSSION TITLE
Update /Set profile picture

### DISCUSSION THREAD

**Discussion by Stein-Hakase** [tulir/whatsmeow#248] [WhatsApp protocol Q&A] [answered]
As described in title , im looking for a way to update and set profile picture  ( i"ve found SetGroupPhoto , but it's not working for updatation own profile picture ).

**Comment by mamur-rezeki** [comment, marked as answer]
Yes, we can use `SetGroupPhoto` with empty jid for updating our own profile picture or group jid for updating group picture.

The size of picture probably limited to 640x640 or less.
It would be returned error if the size of picture didn't match requirements.

**Reply by Stein-Hakase** [reply]
Thank you very much i already figured out how I forgot to update here :
```
DI := meow.Client.DangerousInternals()
  diq := whatsmeow.DangerousInfoQuery{
        Namespace: "w:profile:picture",
        Type: "set",
        To: types.ServerJID,
        Content: []binary.Node{{
        Tag: "picture",
        Attrs: binary.Attrs{"type": "image"},
        Content: buffImg.Bytes(),
  }},
}
_, err = DI.SendIQ(diq)
```

but your solution is better and cleaner !
if you have any ideas about how to set pushname I will be glad

**Reply by chensien** [reply]
I tried to use this method, but I encountered this error (info query returned status 406: not-acceptable)
